middleware.ts

```diff
@@ -88,7 +88,18 @@ export default async function middleware(req: NextRequest) {
     (locale) => !pathname.startsWith(`/${locale}/`) && pathname !== `/${locale}`
   );
 
-  console.log('isLocaleMissing', isLocaleMissing);
+  const paramLocale = url.searchParams.get('locale');
+  const hasEmbedParam = url.searchParams.has('embed');
+  const pathLocale = url.pathname.split('/')[1];
```

Comment:
Locale may not be present in the path on the initial request (e.g. if the user visits web.plant-for-the-planet/profile)